### ISSUE TITLE
Add-LOBAPP

### ISSUE THREAD

**Issue by ibrahimsharif** [microsoftgraph/powershell-intune-samples#29]
Hi 
Thanks for the new samble script awsome work , 🥇 
but  when i use Upload-AndroidLob  and i look from the azure portal the 

Version code displays the version version name and the version code is an empty field. 

I am using the android sdk 

![image](https://user-images.githubusercontent.com/24541271/37150277-74805064-22d1-11e8-8f62-fa7e63da003b.png)

We look forward to have a patch exitsting lob app ps script sample 🔢   

Best regards from

Ibrahim

**Comment by ibrahimsharif** [comment]
if Version Code or Version name isnt provided we wont be able to Patch  with new .APK app trohug the azure portal  , we will get an error 
![image](https://user-images.githubusercontent.com/24541271/37154363-08767682-22e0-11e8-9962-b7517b292a83.png)

**Comment by davefalkus** [comment]
Hi @ibrahimsharif 

Thanks for spotting this, will fix it shortly and update the current sample.

Dave

**Comment by davefalkus** [comment]
Hi @ibrahimsharif 

I've just updated the script and those values are now added into the service, could you please test and let me know if it now works for you. If it works you can close the issue.

Thanks

Dave

**Comment by ibrahimsharif** [comment]
Hi @davefalkus  

The Values have oppesit possetions 
![image](https://user-images.githubusercontent.com/24541271/37194083-d999ce3a-236c-11e8-9edb-cb9a027afffb.png)

the version code is the 5 digits number and the version name is  2.3.10.0-stable

I Did solve it by

* selecting possition 2 item from the string array  in Get-ApkInformation
  
`
	$PackageInfo[1].Split("'")[1]
	$PackageInfo[3].Split("'")[1]
	$PackageInfo[2].Split("'")[1]
	
	if ($logContent) { Write-Host -ForegroundColor Gray $PackageInfo[1].Split("'")[1]; }
	if ($logContent) { Write-Host -ForegroundColor Gray $PackageInfo[3].Split("'")[1]; }
	if ($logContent) { Write-Host -ForegroundColor Gray $PackageInfo[2].Split("'")[1]; }

`


* Adding $identityVersionCode to GetAndroidAppBody As param and to the body 
`
function GetAndroidAppBody($displayName, $publisher, $description, $filename, $identityName, $identityVersion, $identityVersionCode, $minimumSupportedOperatingSystem)
{
	
	$body = @{ "@odata.type" = "#microsoft.graph.androidLOBApp" };
	$body.categories = @();
	$body.displayName = $displayName;
	$body.publisher = $publisher;
	$body.description = $description;
	$body.fileName = $filename;
	$body.identityName = $identityName;
	$body.identityVersion = $identityVersion;
	$body.versionName = $identityVersion;
	$body.versionCode = $identityVersionCode;
	
	if ($minimumSupportedOperatingSystem -eq $null)
	{
		
		$body.minimumSupportedOperatingSystem = @{ "v4_4" = $true };
		
	}
	
	else
	{
		
		$body.minimumSupportedOperatingSystem = $minimumSupportedOperatingSystem;
		
	}
	
	$body.informationUrl = $null;
	$body.isFeatured = $false;
	$body.privacyInformationUrl = $null;
	$body.developer = "";
	$body.notes = "";
	$body.owner = "";
	
	$body;
}
`

* in the Uploade-AndroidLOB did follwing changes
`
		if (!$identityVersionCode)
		{
			
			Write-Host
			Write-Host "Opening APK file to get identityVersion to pass to the service..." -ForegroundColor Yellow
			
			$APKInformation = Get-ApkInformation -AndroidSDK $AndroidSDKLocation -sourceFile "$SourceFile"
			
			$identityVersionCode = $APKInformation[2]
			
		}

`
`
		
		[xml]$manifestXML = '<?xml version="1.0" encoding="utf-8"?><AndroidManifestProperties xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><Package>com.test.app</Package><PackageVersionCode>1</PackageVersionCode><PackageVersionName>1.0</PackageVersionName><ApplicationName>FileName.apk</ApplicationName><MinSdkVersion>3</MinSdkVersion><AWTVersion></AWTVersion></AndroidManifestProperties>'
		
		$manifestXML.AndroidManifestProperties.Package = "$identityName" # com.application.test
		$manifestXML.AndroidManifestProperties.PackageVersionCode = "$identityVersionCode" # 10
		$manifestXML.AndroidManifestProperties.PackageVersionName = "$identityVersion" # 1.0.5.4
		$manifestXML.AndroidManifestProperties.ApplicationName = "$filename" # name.apk
		
		$manifestXML_Output = $manifestXML.OuterXml.ToString()

`

i dont now if the manifest information VersionCode ,VersionName is used by android when the app gets installed but i thinks its important that the values from manifest are correct

in the current sample the 
VersionName,VersionCode are assigned the values VersionName

![image](https://user-images.githubusercontent.com/24541271/37195260-58e10f5a-2372-11e8-948c-3bc581bbac12.png)


 
Thanks David four quick respons i appreciated it ,you have done an awsome work with the ADD-LOB .PS

/Ibrahim

**Comment by davefalkus** [comment]
Thanks for the great information @ibrahimsharif 

What we did was follow the manual UI upload of a LOB application. If you upload your application by the UI does it add the values in the order your expecting? For the application that we've tested the script add the values in the same place as the UI upload does. I will test with some other applications, but would be good to understand if you see the same behaviour from a UI upload.

Thanks

Dave
